core/io/image.cpp

```diff
@@ -3303,7 +3303,7 @@ void Image::_set_color_at_ofs(uint8_t *ptr, uint32_t ofs, const Color &p_color)
 			uint16_t rgba = 0;
 
 			rgba = uint16_t(CLAMP(p_color.r * 31.0, 0, 31));
-			rgba |= uint16_t(CLAMP(p_color.g * 63.0, 0, 33)) << 5;
+			rgba |= uint16_t(CLAMP(p_color.g * 63.0, 0, 63)) << 5;
```

Comment:
That rang a bell:

https://github.com/godotengine/godot/pull/60135#discussion_r1104278399

Might be worth reviving/taking over that PR if you want, it wasn't merged because I'm not confident it's correct without proper unit tests.